calpoly-csai/virtual-tour-concept
# Convert to typescript

We want to use typescript in this project going forward for backend logic (not components).

## What's new
- [x] Installed typescript and associated dependencies
- [x] Set up config files
- [x] Convert each modules to typescript

## Bakground
After converting a few components to typescript, @kgoo124 and I ran into a ton of annoying and unhelpful errors that slowed development more than it helped it. We decided that it wouldn't help developers to convert our .jsx files to typescript, but that it could still help our backend logic be more clear. Going forward, we will try to put our backend logic (ie data handling) functions in the `src/modules/` file

## QA
Make sure there are no errors when executing `npm run build` (should call the `tsc` command which will compile the typescript).
